src/System.Private.ServiceModel/src/System/ServiceModel/Channels/CommunicationObject.cs

```diff
@@ -798,12 +903,10 @@ internal protected virtual Task OnCloseAsync(TimeSpan timeout)
 
         internal protected virtual Task OnOpenAsync(TimeSpan timeout)
```

Comment:
Reviewers: doing a code coverage run reveals this and the virtual OnCloseAsync here in the base class now appears to be dead code.  This is consistent with all WCF product code overriding this method, plus the fact the way we call external types doesn't use the code path that gets here.  I'm contemplating a hard assert here that we should never get here, and then remove the body of these methods.  Thoughts?